packages/renderer/src/lib/dialogs/MessageBoxUtils.ts

```diff
@@ -0,0 +1,31 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+export function withConfirmation(func: () => unknown, action: string): void {
+  window
+    .showMessageBox({
+      title: 'Confirmation',
+      message: 'Are you sure you want to ' + action + '?',
+      buttons: ['Yes', 'Cancel'],
+    })
+    .then(result => {
+      if (result && result.response === 0) {
+        func();
+      }
+    });
+}
```

Comment:
I expect to have something being really asynchronous and resolving when user clicked

Reply:
Hummm, but it should go inside the `on:click` therefore cannot return a promise ? I am not sure to follow what you expect

Reply:
cc @benoitf

Reply:
Another solution could be to have withConfirmation(callback: (err?: unknown) => void).

If you want to have it with an async you can do the following

```
import { promisify } from 'node:util'

const asyncConfirmation = promisify(withConfirmation)
```

what do you think ?

Reply:
> Why do you want to introduce callback when the code base is using async ?

Because you want to be called back, after the confirmation dialog, on success.

Reply:
then return a Promise<boolean>

Reply:
I added documentation to clarify usage